internal/app/machined/pkg/runtime/v1alpha2/v1alpha2_controller.go

```diff
@@ -87,6 +89,44 @@ func (ctrl *Controller) Run(ctx context.Context, drainer *runtime.Drainer) error
 		return err
 	}
 
+	var (
+		etcRoot        xfs.Root
+		networkEtcRoot xfs.Root
+	)
+
+	opentreeOnAnonymous, err := xfs.HasKernel("6.15.0")
```

Comment:
I think we should put this under a check that we are in a container mode, as if we run with Talos kernel, we should never try to fallback to non-secure tmpfs